Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -234,6 +233,7 @@ - (void)timeout:(NSTimer *)timer
                                                         selector:@selector(timeout:)
                                                         userInfo:request
                                                          repeats:NO];
+  _pendingRequests = [NSMutableArray new];
```

Comment:
I think this needs to be

    if (!_pendingRequests) {
      _pendingRequests = [NSMutableArray new];
    }

Otherwise you're wiping out all previous requests each time you call getCurrentPosition().